### DISCUSSION TITLE
Connect VSCode to a custom ballerina pack

### DISCUSSION THREAD

**Discussion by manuranga** [ballerina-platform/ballerina-lang#40311] [Q&A] [answered]
I tried setting ballerina home from VSCode settings, bit it fails with `ballerina: unknown command 'start-language-serve'`.
This is not surprising since if I run `bal start-language-serve` in CLI it also prints `[Error - 11:54:04] Ballerina LS Client client: couldn't create connection to server.
Launching server using command ~/Downloads/ballerina-2201.5.0-swan-lake/bin/bal failed. Error: spawn ~/Downloads/ballerina-2201.5.0-swan-lake/bin/bal ENOENT
Failed to activate Ballerina extension. Error when checking ballerina version. undefine`. This is the same command vscode is trying to run.

How can I connect VSCode to a custom ballerina pack?

**Comment by gigara** [comment]
@IMS94 @malinthar

**Comment by manuranga** [comment, marked as answer]
Looks like my issue is resolved, I was setting home as `~/Downloads/ballerina-2201.5.0-swan-lake/bin/bal` but it should have been `/Users/manurangaperera/Downloads/ballerina-2201.5.0-swan-lake/bin/bal`.
